staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -80,3 +118,23 @@ type resultAdapter struct {
 func (r *resultAdapter) Increment(code, method, host string) {
 	r.m.WithLabelValues(code, method, host).Inc()
 }
+
+type expirationAdapter struct {
+	m *k8smetrics.Gauge
+}
+
+func (e *expirationAdapter) Set(t time.Time) {
+	if t.IsZero() {
+		e.m.Set(0)
```

Comment:
Maybe don't report this at all?
Let the metric be empty if there are not certs, instead of reporting fake time value.

Reply:
I looked into the Gauge implementation of Prometheus. Its default value when read is a float64 of 0, there is no "empty" value. Basically this is just resetting it to its empty value. I figured this is preferred when somehow a client cert refreshes into only a token (no expr), we shouldn't leave this metric as though that old cert is just sitting around not being rotated.

Also I considered setting m.Set(math.Inf(1)) on a zero time, but that would require setting the Gauge to that value at creation time which I feel would be pushing all kinds of implementation details to places they shouldn't be \*_shrug_\*.